src/controllers/hooks/router-store/index.tsx

```diff
@@ -18,3 +31,43 @@ export const useRouterActions = (): RouterActionsType => {
 
   return allActions;
 };
+
+const useParam = createHook<
+  EntireRouterState,
+  AllRouterActions,
+  string,
+  { paramKey: string }
+>(RouterStore, {
+  selector: ({ query }, { paramKey }): string => query[paramKey],
+});
+
+/**
+ * Utility hook for accessing URL query params
+ */
+export const useQueryParam = (
+  paramKey: string
+): [string | undefined, (qp: string | null) => void] => {
+  const [paramVal, routerActions] = useParam({ paramKey });
+
+  const setQueryParam = React.useCallback(
+    (newValue: string | undefined | null) => {
+      const {
+        location: { pathname },
+        query,
+      } = getRouterState();
```

Comment:
@albertogasparin Is it a good idea to access the store with `getRouterState` here? We need the `query` object for non-reactive read only purpose here. If there's a better approach lemme know pl.

Reply:
No, I don't think it is the best approach. I think it could be much better to move all this query (and path) manipulations to a sweets-state action, so we don't have to make the entire store available in the hook (`getRouterState` is an anti pattern)

We can either create specific `pushQuery`/`pushPath` actions to manipulate the url and call the generic `push`, or we can extend `push` directly and allow it to be called with an object. Something like:

```js
push({ param: { key: value }, query: { key: value }  });
```

The good thing about the latter is that we can in future allow using push without requiring string manipulation from the consumer side, as the API could evolve to allow:

```js
push({ route, param: { key: value }, query: { key: value }  });
// also replace could allow such shape
replace({ route, param: { key: value }  });
```

The main thing that we should pay attention at is how do we communicate from the API when `param/query` keys are merged with existing ones or replaced. We might need a specific option to make it obvious.

What I mean is, assuming we have `param: { foo: 1, bar: 1 }`, how do we allow consumers to say with `push({ param: { foo: 2 } })` that they want as a result `param: { foo: 2, bar: 1 }` or just `param: { foo: 2 }`. 

Maybe something like `push({ param: { ... }, query: { ... }, merge: true })` ?
Or `partial: true`? Or `override: false`? Or ideas?

Reply:
@albertogasparin I'm glad I pointed this out. I've now removed `getRouterState` and created two new actions `pushQueryParam` and `pushPathParam` that does the job for the hooks. 

Re: extending the push action to support query and path params as API
I looked into this a bit and figured there are a few moving parts that need to be considered before exposing it as an api (eg. absolute URLs and others) and it leads to more code paths. I think we could revisit this when we decide if we are going to expose push action to support a config param.